pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -271,11 +230,6 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 		}
 	}
 
-	// This is hack to keep consistent with previous behavior.
-	if listenerPort != 80 {
-		// only select virtualServices that matches a service
-		virtualServices = selectVirtualServices(virtualServices, servicesByName)
```

Comment:
`IstioEgressListenerWrapper.VirtualServices` is not equal to this selection, this is port related

Reply:
I added that logic inside filterVirtualServices

Reply:
https://github.com/istio/istio/blob/77366e7a2b077d426df043669f09fe23d4a22c7e/pilot/pkg/networking/core/v1alpha3/httproute.go#L254-L277 

You can see here it selects the services that has specific port, but i do not see there is port related logic in sidecar